### PR TITLE
feat: sharding-e2e-tests

### PR DESCRIPTION
**Pull request by charles-testco**
## Reason for Change

- #TICKET_NUMBER
- If the reason for this PR's code changes are not clear in the issue, state value/impact

## Changes

- add
- remove
- modify

## Testing steps

- Either list QA steps or reasoning you feel QA is unnecessary
- Describe how you made sure to know that your changes worked. Should allow someone else to go verify your code without in depth knowledge.
- "Unit tested only", "tested in rdev by a, b, c, verifying feature worked by... ", "manually ran pipeline locally with these results: ..."

### CODE REVIEW

File: frontend/package.json [hunk 1]
```diff
@@ -94,6 +94,8 @@
     "next-dev-https": "0.1.1",
     "next-mdx-remote": "^4.1.0",
     "playwright": "~1.32.0",
+    "playwright-merge-html-reports": "^0.2.8",
+    "react": "^17.0.2",
```
Comment (tihuan) [inline frontend/package.json:98]: we can remove this since it's already in `dependencies`!

**Review by tihuan — COMMENTED** [review]
Thanks for the PR, Charles! Looks like the e2e-test is still waiting for status report 🤔 

PTAL thank you!

File: frontend/package-lock.json
Comment (tihuan) [inline frontend/package-lock.json:1]: maybe remove this file, since we didn't update dependencies! Thank you!

**Comment by charles-testco** [comment]
> Thanks for the PR, Charles! Looks like the e2e-test is still waiting for status report 🤔
> 
> PTAL thank you!

This is frustrating. I have switched the name a bit but still no luck. Will have to read more.

**Comment by tihuan** [comment]
@ebezzi this is the PR that shards tests! 

In the screenshot, you can see that the PR check `e2e-test` is still waiting for different shards to report their outcomes 🤔 
Screenshot 2023-06-14 at 1 41 56 PM

Thank you!

**Comment by charles-testco** [comment]
> @ebezzi this is the PR that shards tests!
> 
> In the screenshot, you can see that the PR check `e2e-test` is still waiting for different shards to report their outcomes 🤔 Screenshot 2023-06-14 at 1 41 56 PM
> 
> Thank you!

@tihuan I think I know what the problem is; just need to figure how to fix it. The below logic is not correct and is skipped if any of the shards fail, but what we want is to wait for all shards to complete whether they fail or pass.

`wait-for-shards:
    name: Wait for shards
    needs: run-e2e-tests # run after shards
    runs-on: ubuntu-20.04
    if: success() # only run when all shards have passed
    # store success output flag for ci job
    outputs:
      success: ${{ steps.setoutput.outputs.success }}
    steps:
      - id: setoutput
        run: echo "{success}={true}" >> $GITHUB_OUTPUT`

**Comment by charles-testco** [comment]
@tihuan I think it is working, no?

**Comment by tihuan** [comment]
Amazing 🔥 `Push Tests / e2e-test (pull_request)` failed! Thanks so much for testing this out, @charles-testco 🙏 Really appreciate it!

Please feel free to revert the dummy test and merge the PR!

**Review by tihuan — APPROVED** [review]
LGTM after reverting the dummy test 🙏 Thanks so much again for making this happen!

**Comment by charles-testco** [comment]
> Amazing 🔥 `Push Tests / e2e-test (pull_request)` failed! Thanks so much for testing this out, @charles-testco 🙏 Really appreciate it!
> 
> Please feel free to revert the dummy test and merge the PR!

Will do. Thanks. We will have have the question of merging html reports but I suppose we are using the Allure more here and should not be a blocker?

**Comment by tihuan** [comment]
Yeah I'm assuming the shards will upload their results to the same place for Allure to aggregate, so it should be okay. But let's confirm that's what Allure is doing too to be sure lol

**Comment by tihuan** [comment]
Hey @charles-testco ! Sorry it seems like the checks are no longer passing 😭 PTAL thank you!

**Comment by tihuan** [comment]
porting solution over to https://github.com/chanzuckerberg/single-cell-data-portal/pull/5492. Thanks @charles-testco !

**Pull request closed without merging**